File: LAB2/Calkowanie.java
import java.util.Scanner;

public class Calkowanie {
    public static void main(String[] args) {
        double start, stop, dx;
        int n;
        Scanner scanin = new Scanner(System.in);
        System.out.println("Podaj przedzial calkowania: ");
        start = scanin.nextDouble(); //xp
        System.out.println("Podaj koniec przedzialu calkowania: ");
        stop = scanin.nextDouble(); // xk
        System.out.println("Podaj dokladnosc calkowania(n):");
        n = scanin.nextInt();
        System.out.println(start + " " + stop +" "+n);
        dx = (stop - start) / (double)n;
        System.out.println("Wybierz metodę:\n0-Simpson\n1-Prostokatow\n2-Trapezow");
        int metoda = scanin.nextInt();
        switch (metoda){
            case 0:
                M_Simpson klasa = new M_Simpson(start, stop, dx);
                try{
                    for (int a=1;a<n;a++){
                        Thread ai = new M_Simpson(a);
                        ai.start();
                        ai.join();
                    }
                }catch (InterruptedException ex){}
                klasa.licz();
                break;
            case 1:
                M_prostokatow klas = new M_prostokatow(start, stop, dx);
                try{
                    for (int a=1;a<=n;a++){
                        Thread ai = new M_prostokatow(a);
                        ai.start();
                        ai.join();
                    }
                }catch (InterruptedException ex){}
                klas.licz();
                break;
            case 2:
                M_trapezow kla = new M_trapezow(start, stop, dx);
                try{
                    for (int a=1;a<n;a++){
                        Thread ai = new M_trapezow(a);
                        ai.start();
                        ai.join();
                    }
                }catch (InterruptedException ex){}
                kla.licz();
                break;
        }
    }
}

class M_Simpson extends Thread{
    static double xk, xp, dx, calka, s;
    int n;
    double x;

    static double func(double x) {return x * x + 3;}
    public M_Simpson(double start, double stop, double dx){
        xk=stop;
        xp=start;
        this.dx = dx;
        s=func(xk - dx/2);
        calka=0;
    }
    public M_Simpson(int n){this.n=n;}

    public void licz(){
        calka = (dx/6) * (func(xp) + func(xk) + 2*calka + 4*s);
        System.out.println("Wynik = "+calka);
    }

    public void run(){
        x = xp + n*dx;
        s += func(x - dx / 2);
        calka += func(x);
        System.out.println("Wątek: " + Thread.currentThread().getId()+" : "+calka);
    }
}

class M_prostokatow extends Thread {
    static double xk, xp, dx, calka;
    int n;

    static double func(double x) {return x * x + 3;}
    public M_prostokatow(int n){this.n=n;}

    public M_prostokatow(double start, double stop, double dx){
        xk=stop;
        xp=start;
        this.dx = dx;
    }

    public void licz(){
        calka *= dx;
        System.out.println("Wynik = "+calka);
    }

    public void run(){
        calka += func(xp + n * dx);
        System.out.println("Wątek: "+n+" " + Thread.currentThread().getId()+" : "+calka);
    }
}

class M_trapezow extends Thread {
    static double xk, xp, dx, calka;
    int n;

    static double func(double x) {return x * x + 3;}

    public M_trapezow(int n){this.n=n;}

    public M_trapezow(double start, double stop, double dx){
        xk=stop;
        xp=start;
        this.dx = dx;
        calka=(func(xp) + func(xk)) / 2;
    }

    public void licz(){
        calka *= dx;
        System.out.println("Wynik = "+calka);
    }

    public void run(){
        calka += func(xp + n * dx);
        System.out.println("Wątek: " + Thread.currentThread().getId()+" : "+calka);
    }
}

File: LAB10/Calkowanie.cpp
#include <stdio.h>
#include <stdlib.h>
#include <chrono>
#include <iostream> 
using namespace std;

double func(float x)
{
	return 3*x*x-6*x+5;
}

double M_Trapezow(float a, float b, int n)
{
    double calka = 0;
    float dx = (b-a)/n;

    for (int i=1; i<=n-1; i++)
    {
	calka += func(a+i*dx);
    }
	
	calka += (func(a)+func(b))/2;
	calka *= dx;
    return calka;
}

double M_Prostokatow(float xp, float xk, int n)
{
    double calka = 0;
    double dx = (xk - xp) / (float)n;
 
    for (int i=1; i<=n; i++) {
    calka += func(xp + i * dx);
    }
    calka *= dx;
    return calka;
}

double M_Simpson(float xp, float xk, int n)
{
    double calka = 0;
    double s, x;
    int i;
    double dx = (xk - xp) / (float)n;

    s = 0;
    for (i=1; i<n; i++) {
        x = xp + i*dx;
        s += func(x - dx / 2);
        calka += func(x);
    }
    s += func(xk - dx / 2);
    calka = (dx/6) * (func(xp) + func(xk) + 2*calka + 4*s);
    
    return calka;
}

int main(int argc, char **argv)
{
    int xp,xk,n;
    // printf("Podaj poczatek przedzialu calkowania\n");
    // scanf("%f", &xp);
    
    // printf("Podaj koniec przedzialu calkowania\n");
    // scanf("%f", &xk);

    xp=1;
    xk=2;
    // printf("Ilosc procesow: "); 
    // int procesy;
    // scanf("%d", &procesy);

    printf("Podaj dokladnosc calkowania\n");
    scanf("%d", &n);

    printf("Przedzial xp: %d, xk: %d\n", xp, xk);
    double trapezy, prostokaty, simpson;
    ///Trapezy
    auto start2 = chrono::steady_clock::now();
    trapezy = M_Trapezow(xp, xk, n);
    auto stop = chrono::steady_clock::now();
    auto diff = stop - start2;
    cout << "Wynik metoda trapezow: " << trapezy << " :" << chrono::duration <double, milli> (diff).count() << " ms" << endl;
    //Prostokaty
    start2 = chrono::steady_clock::now();
    prostokaty = M_Prostokatow(xp, xk, n);
    stop = chrono::steady_clock::now();
    diff = stop - start2;
    cout << "Wynik metoda prostokatow: " << prostokaty << " :" << chrono::duration <double, milli> (diff).count() << " ms" << endl;
    //Simpson
    start2 = chrono::steady_clock::now();
    simpson = M_Simpson(xp, xk, n);
    stop = chrono::steady_clock::now();
    diff = stop - start2;
    cout << "Wynik metoda simpsona: " << simpson << " :" << chrono::duration <double, milli> (diff).count() << " ms" << endl;
}
